developers/creating-your-first-plugin.rst

```diff
@@ -192,3 +192,80 @@ Velocity automatically registers your plugin main class as a listener.
             // For instance, we could register an event:
             server.getEventManager().register(this, new PluginListener());
         }
+
+Alternatively: Kotlin
+^^^^^^^^^^^^^^^^^^^^^
+
+Kotlin is the second-most popular language capable of running on the JVM (behind Java).
+As such, it's a viable option for developing Velocity plugins; additionally, its more
+modern syntax makes it a very attractive choice.
+
+Here are some brief notes to assist in starting a Velocity plugin using Kotlin:
+
+Your build script will look a little different than the above, because of the additional
+dependency on the Kotlin stdlib and a specialized annotation processor that must be used
+in place of Gradle's default. Also, it will probably be written in Kotlin itself, rather
+than Groovy.
+
+Your ``build.gradle.kts`` should include the following:
+
+.. code-block:: kotlin
+
+    plugins {
+        kotlin("jvm") version "1.4.10"
+        kotlin("kapt") version "1.4.10"
+        id("com.github.johnrengelman.shadow") version "6.0.0"
+    }
+
+    repositories {
+        mavenCentral()
+        maven {
+            url = uri("https://repo.velocitypowered.com/snapshots/")
+        }
+    }
+
+    var velocityApi = "com.velocitypowered:velocity-api:1.0.9"
+
+    dependencies {
+        implementation(kotlin("stdlib"))
+        implementation(velocityApi)
+        kapt(velocityApi)
+    }
+    
+    tasks["build"].dependsOn("shadowJar")
```

Comment:
This feels like something subjective; not everyone wants this, and the shadowJar task does still exist.

Reply:
There's a practical reason: Velocity doesn't contain the Kotlin runtime, primarily because Velocity is not written in Kotlin.

Reply:
Right; the build is unusable without the shadowJar